Mage.Sets/src/mage/cards/h/HopelessNightmare.java

```diff
@@ -0,0 +1,48 @@
+package mage.cards.h;
+
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.PutIntoGraveFromBattlefieldSourceTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.common.LoseLifeOpponentsEffect;
+import mage.abilities.effects.common.SacrificeSourceEffect;
+import mage.abilities.effects.common.discard.DiscardEachPlayerEffect;
+import mage.abilities.effects.keyword.ScryEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.TargetController;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class HopelessNightmare extends CardImpl {
+
+    public HopelessNightmare(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{B}");
+
+
+        // When Hopeless Nightmare enters the battlefield, each opponent discards a card and loses 2 life.
+        Ability ability = new EntersBattlefieldTriggeredAbility(new DiscardEachPlayerEffect(TargetController.OPPONENT));
+        ability.addEffect(new LoseLifeOpponentsEffect(2).setText("loses 2 life").concatBy("and"));
```

Comment:
you don't need `concatBy` if you're changing text but it's fine here

Reply:
Oh sure, I did start with only concatBy, then realised that there was a redundant "each opponent".